data-loader/run_data_loader.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+python="/home/data/apps/topfibers/repo/environments/env_code/bin/python"
+
+loader_home="/home/data/apps/topfibers/repo/data-loader"
+
+# Create SSH tunnel
+ssh -i /u/truthy/.ssh/id_rsa -4 -N -L 5580:localhost:5432 truthy@lisa.luddy.indiana.edu &
+
+# Store the PID of the SSH tunnel process
+tunnel_pid=$!
```

Comment:
ooh what does `$!` do?

I can't find anything about this?

What I know if is:

| Variable | Use |
| -- | -- |
| $# | Stores the number of command-line arguments that were passed to the shell program. |
| $? | Stores the exit value of the last command that was executed. |
| $0 | Stores the first word of the entered command (the name of the shell program). |
| $* | Stores all the arguments that were entered on the command line ($1 $2 ...). |
| "$@" | Stores all the arguments that were entered on the command line,    individually quoted ("$1" "$2" ...). |

from http://linuxsig.org/files/bash_scripting.html

Reply:
I am not sure this is the best method, but this works as $! will keep the background process id. Here some links, that it has, https://www.cyberciti.biz/faq/how-to-return-pid-of-a-last-command-in-linux-unix/ ,  https://stackoverflow.com/questions/29340087/bash-store-command-pid-in-variable-and-kill-process
Any suggestions to make this..